src/cmap/connect.ts

```diff
@@ -329,14 +329,13 @@ const SOCKET_ERROR_EVENTS = new Set(SOCKET_ERROR_EVENT_LIST);
 function makeConnection(options: MakeConnectionOptions, _callback: Callback<Stream>) {
   const useTLS = options.tls ?? false;
   const keepAlive = options.keepAlive ?? true;
-  const socketTimeoutMS = options.socketTimeoutMS ?? Reflect.get(options, 'socketTimeout') ?? 0;
   const noDelay = options.noDelay ?? true;
   const connectTimeoutMS = options.connectTimeoutMS ?? 30000;
   const rejectUnauthorized = options.rejectUnauthorized ?? true;
-  const keepAliveInitialDelay =
-    ((options.keepAliveInitialDelay ?? 120000) > socketTimeoutMS
-      ? Math.round(socketTimeoutMS / 2)
-      : options.keepAliveInitialDelay) ?? 120000;
+  // Default to delay to 300 seconds. Node automatically then sets TCP_KEEPINTVL to 1 second
+  // which is acceptable to the recommendation of 10 seconds and also cannot be configured.
+  // TCP_KEEPCNT is also set to 10 in Node and cannot be configured. (Recommendation is 9)
+  const keepAliveInitialDelay = options.keepAliveInitialDelay || 300000;
```

Comment:
```suggestion
  const keepAliveInitialDelay = options.keepAliveInitialDelay;
```

Because of the defaulting, we should always have a defined number here.

Reply:
Updated.